### PR TITLE
Validation remove and change in files related to curve in library

### PR DESCRIPTION
**Pull request by tuvshinzayaArm**
## Description

This is part of #4681 
This PR re-introduce enum check in files related to curve in library.
Additionally, removes NULL pointer checks ( That is already disabled by #4313 )

## Status
**IN DEVELOPMENT**

## Requires Backporting

NO  

## Migrations

NO

## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Changelog updated
- [ ] Backported

### CODE REVIEW

File: library/ecdsa.c [hunk 1]
```diff
@@ -404,12 +398,8 @@ int mbedtls_ecdsa_sign( mbedtls_ecp_group *grp, mbedtls_mpi *r, mbedtls_mpi *s,
                 const mbedtls_mpi *d, const unsigned char *buf, size_t blen,
                 int (*f_rng)(void *, unsigned char *, size_t), void *p_rng )
 {
-    ECDSA_VALIDATE_RET( grp   != NULL );
-    ECDSA_VALIDATE_RET( r     != NULL );
-    ECDSA_VALIDATE_RET( s     != NULL );
-    ECDSA_VALIDATE_RET( d     != NULL );
-    ECDSA_VALIDATE_RET( f_rng != NULL );
-    ECDSA_VALIDATE_RET( buf   != NULL || blen == 0 );
+    if( buf == NULL && blen != 0 )
```
Comment (AndrzejKurek) [inline library/ecdsa.c:401]: If the issue description is correct - this could be removed.

File: library/ecdsa.c [hunk 2]
```diff
@@ -503,13 +493,8 @@ int mbedtls_ecdsa_sign_det_ext( mbedtls_ecp_group *grp, mbedtls_mpi *r,
...
                                 void *p_rng_blind )
 {
-    ECDSA_VALIDATE_RET( grp   != NULL );
-    ECDSA_VALIDATE_RET( r     != NULL );
-    ECDSA_VALIDATE_RET( s     != NULL );
-    ECDSA_VALIDATE_RET( d     != NULL );
-    ECDSA_VALIDATE_RET( buf   != NULL || blen == 0 );
-    ECDSA_VALIDATE_RET( f_rng_blind != NULL );
-
+    if( buf == NULL && blen != 0 )
```
Comment (AndrzejKurek) [inline library/ecdsa.c:496]: If the issue description is correct - this could be removed.

File: library/ecdsa.c [hunk 3]
```diff
@@ -634,11 +619,8 @@ int mbedtls_ecdsa_verify( mbedtls_ecp_group *grp,
                           const mbedtls_mpi *r,
                           const mbedtls_mpi *s)
 {
-    ECDSA_VALIDATE_RET( grp != NULL );
-    ECDSA_VALIDATE_RET( Q   != NULL );
-    ECDSA_VALIDATE_RET( r   != NULL );
-    ECDSA_VALIDATE_RET( s   != NULL );
-    ECDSA_VALIDATE_RET( buf != NULL || blen == 0 );
+    if( buf == NULL && blen != 0 )
```
Comment (AndrzejKurek) [inline library/ecdsa.c:622]: If the issue description is correct - this could be removed.

File: library/ecjpake.c [hunk 4]
```diff
@@ -107,10 +99,10 @@ int mbedtls_ecjpake_setup( mbedtls_ecjpake_context *ctx,
 {
     int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
 
-    ECJPAKE_VALIDATE_RET( ctx != NULL );
-    ECJPAKE_VALIDATE_RET( role == MBEDTLS_ECJPAKE_CLIENT ||
-                          role == MBEDTLS_ECJPAKE_SERVER );
-    ECJPAKE_VALIDATE_RET( secret != NULL || len == 0 );
+    if( role != MBEDTLS_ECJPAKE_CLIENT && role != MBEDTLS_ECJPAKE_SERVER )
+        return( MBEDTLS_ERR_ECP_BAD_INPUT_DATA );
+    if( secret == NULL && len != 0 )
```
Comment (AndrzejKurek) [inline library/ecjpake.c:104]: If the issue description is correct - this could be removed.

File: library/ecp.c [hunk 5]
```diff
@@ -737,12 +705,8 @@ int mbedtls_ecp_point_write_binary( const mbedtls_ecp_group *grp,
 {
     int ret = MBEDTLS_ERR_ECP_FEATURE_UNAVAILABLE;
     size_t plen;
-    ECP_VALIDATE_RET( grp  != NULL );
-    ECP_VALIDATE_RET( P    != NULL );
-    ECP_VALIDATE_RET( olen != NULL );
-    ECP_VALIDATE_RET( buf  != NULL );
-    ECP_VALIDATE_RET( format == MBEDTLS_ECP_PF_UNCOMPRESSED ||
-                      format == MBEDTLS_ECP_PF_COMPRESSED );
+    if( format != MBEDTLS_ECP_PF_UNCOMPRESSED && format != MBEDTLS_ECP_PF_COMPRESSED )
```
Comment (AndrzejKurek) [inline library/ecp.c:708]: Line too long - please break.

File: library/ecp.c [hunk 6]
```diff
@@ -911,12 +866,8 @@ int mbedtls_ecp_tls_write_point( const mbedtls_ecp_group *grp, const mbedtls_ecp
                          unsigned char *buf, size_t blen )
 {
     int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
-    ECP_VALIDATE_RET( grp  != NULL );
-    ECP_VALIDATE_RET( pt   != NULL );
-    ECP_VALIDATE_RET( olen != NULL );
-    ECP_VALIDATE_RET( buf  != NULL );
-    ECP_VALIDATE_RET( format == MBEDTLS_ECP_PF_UNCOMPRESSED ||
-                      format == MBEDTLS_ECP_PF_COMPRESSED );
+    if( format != MBEDTLS_ECP_PF_UNCOMPRESSED && format != MBEDTLS_ECP_PF_COMPRESSED )
```
Comment (AndrzejKurek) [inline library/ecp.c:869]: Line too long - please break.

**Review by AndrzejKurek — CHANGES_REQUESTED** [review]
Missing tests for:
- `mbedtls_ecdh_get_params` - bad `side`;
- `mbedtls_ecjpake_setup` - bad `role`;
- `mbedtls_ecp_point_write_binary` - bad `format`;
- `mbedtls_ecp_tls_write_point` - bad `format`;
If I'm not mistaken, some other checks should not be ported.

File: library/ecdsa.c (hunk 2 above)
Comment (tuvshinzayaArm) [reply library/ecdsa.c]: removed it

File: library/ecdsa.c (hunk 3 above)
Comment (tuvshinzayaArm) [reply library/ecdsa.c]: removed it

File: library/ecjpake.c (hunk 4 above)
Comment (tuvshinzayaArm) [reply library/ecjpake.c]: removed it

File: library/ecp.c (hunk 5 above)
Comment (tuvshinzayaArm) [reply library/ecp.c]: broke it

File: library/ecp.c (hunk 6 above)
Comment (tuvshinzayaArm) [reply library/ecp.c]: broke it

**Comment by tuvshinzayaArm** [comment]
> Missing tests for:

I think tests for those already existed. I just needed to activated by changing dependencies of test from undefined to no dependency.
Let me know if i understood it wrong.

File: tests/suites/test_suite_ecp.function [hunk 7]
```diff
@@ -61,7 +61,7 @@ inline static int mbedtls_ecp_group_cmp( mbedtls_ecp_group *grp1,
  * END_DEPENDENCIES
  */
 
-/* BEGIN_CASE depends_on:NOT_DEFINED */
+/* BEGIN_CASE depends_on */
```
Comment (AndrzejKurek) [inline tests/suites/test_suite_ecp.function:64]: Please remove the `depends_on` part

**Review by AndrzejKurek — CHANGES_REQUESTED** [review]
I have checked locally that the added enum checks are triggered by invalid param tests.
While the contents of this PR are now almost finished, It would be great to improve the form of commits too:
- If you remove something by accident and later re-introduce it, it's better to rewrite history and to clean up the commits and make reviewing easier, if someone proceeds commit-by-commit. If the code has already been reviewed by someone, this can be postponed to the point when there's already an approval and the history rewriting gives a zero diff compared to before.
- Try to be specific in comment messages - avoid "some" and vague statements;
- Use imperative mood in the subject;

I can recommend reading https://www.freecodecamp.org/news/how-to-write-better-git-commit-messages/ to improve the style a bit. Other than that - PR looks almost finished, good work :)

**Comment by tuvshinzayaArm** [comment]
> ther than that - PR looks almost finished, good work :)

Thanks! i think i will rewrite history now. and push it again.

**Comment by tuvshinzayaArm** [comment]
> I have checked locally that the added enum checks are triggered by invalid param tests. While the contents of this PR are now almost finished, It would be great to improve the form of commits too:
> 
> * If you remove something by accident and later re-introduce it, it's better to rewrite history and to clean up the commits and make reviewing easier for someone proceeding commit-by-commit. If the code has already been reviewed by someone, this can be postponed to the point when there's already an approval and the history rewriting gives a zero diff compared to before.
> * Try to be specific in comment messages - avoid "some" and vague statements;
> * Use imperative mood in the subject;
> 
> I can recommend reading https://www.freecodecamp.org/news/how-to-write-better-git-commit-messages/ to improve the style a bit. Other than that - PR looks almost finished, good work :)

Hi, what do you think about current PR? if it is good. I will change all other related PRs to be like this one.

File: library/ecp.c (hunk 5 above)
Comment (AndrzejKurek) [reply library/ecp.c:708]: Too long again after commit restructuring. Please make sure that the diff between the state from before restructuring and after is zero, as now the reviewers don't have the access to previous state of the code.

File: library/ecp.c (hunk 6 above)
Comment (AndrzejKurek) [reply library/ecp.c:869]: Same as above.

**Review by AndrzejKurek — CHANGES_REQUESTED** [review]
Two long lines came back.

**Comment by tuvshinzayaArm** [comment]
> Two long lines came back.

Fixed it. Btw what coding style rule do we follow?

**Comment by AndrzejKurek** [comment]
> > Two long lines came back.
> 
> Fixed it. Btw what coding style rule do we follow?

https://github.com/Mbed-TLS/mbedtls/blob/development/CONTRIBUTING.md#coding-standards mentions https://tls.mbed.org/kb/development/mbedtls-coding-standards but I can see that it's broken, so we'll use the time machine:
https://web.archive.org/web/20210728231309/https://tls.mbed.org/kb/development/mbedtls-coding-standards

> Mbed TLS source code files use 4 spaces for indentation, not tabs, with a preferred **maximum line length of 80 characters.**

**Comment by tuvshinzayaArm** [comment]
> > > Two long lines came back.
> > 
> > 
> > Fixed it. Btw what coding style rule do we follow?
> 
> https://github.com/Mbed-TLS/mbedtls/blob/development/CONTRIBUTING.md#coding-standards mentions https://tls.mbed.org/kb/development/mbedtls-coding-standards but I can see that it's broken, so we'll use the time machine: https://web.archive.org/web/20210728231309/https://tls.mbed.org/kb/development/mbedtls-coding-standards
> 
> > Mbed TLS source code files use 4 spaces for indentation, not tabs, with a preferred **maximum line length of 80 characters.**

Thanks for review and links!

**Review by AndrzejKurek — APPROVED** [review]
LGTM :)

**Review by wernerlewis — APPROVED** [review]
LGTM

**Pull request merged**